MDANSE_GUI/Src/MDANSE_GUI/InputWidgets/AtomSelectionWidget.py

```diff
@@ -89,10 +139,13 @@ def __init__(
         super().__init__(parent, *args, **kwargs)
         self.setWindowTitle(self._helper_title)
 
-        self.selector = selector
+        self.trajectory = traj_data[1].trajectory
+        self.system = self.trajectory.chemical_system
+        self.selection_model = SelectionModel(self.trajectory)
         self._field = field
-        self.settings = self.selector.settings
-        self.atm_full_names = self.selector.system.name_list
+        self.atm_full_names = self.system.name_list
+        self.molecule_names = [str(x) for x in self.system._clusters.keys()]
+        self.labels = [str(x) for x in self.system._labels.keys()]
```

Comment:
Could be:
```suggestion
        self.molecule_names = list(map(str, self.system._clusters))
        self.labels = list(map(str, self.system._labels))
```

Reply:
I will note you keep externally accessing these variables which are marked `_private`, suggest making them public.